### PR TITLE
Using `DotNetPublish` to drive publishing

### PR DESCRIPTION
**Pull request by jcagme**
This is the change on the Sdk side of things. A follow up PR will come containing setting the variable and upgrading the Sdk version.

### CODE REVIEW

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj [hunk 1]
```diff
@@ -209,7 +206,7 @@
     <MSBuild Projects="Publish.proj"
              Properties="@(_PublishProps)"
              Targets="Publish"
-             Condition="'$(Publish)' == 'true'"/>
+             Condition="'$(Publish)' == 'true' and '$(DotNetPublish)' == 'true'"/>
```
Comment (chcosta) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj]: You have to say `Publish=true` and ``DotNetPublish=true`?  That seems weird to me.

**Comment by mmitche** [comment]
@jcagme , we should probably change the templates and .vsts-ci.yml file in this PR too.

**Comment by jcagme** [comment]
> @jcagme , we should probably change the templates and .vsts-ci.yml file in this PR too.

If we use `-publish` there is no required change (for this particular change that is). Also, if I were to change these files, I think the build would fail since we would be setting things that the "old" sdk does not know about, right?

**Comment by mmitche** [comment]
@jcagme Oh oh right, yeah we'll update this in the SDK update PR.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj [hunk 2]
```diff
@@ -113,9 +115,9 @@
 
     <ItemGroup>
       <_PublishProps Include="@(_CommonProps)"/>
-      <_PublishProps Include="AzureFeedUrl=$(DotNetPublishBlobFeedUrl)" />
-      <_PublishProps Include="AzureAccountKey=$(DotNetPublishBlobFeedKey)" />
-      <_PublishProps Include="DotNetOutputBlobFeedDir=$(_DotNetOutputBlobFeedDir)" />
+      <_PublishProps Include="AzureFeedUrl=$(DotNetPublishBlobFeedUrl)" Condition="$(_PublishToBlobStorage)" />
```
Comment (chcosta) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj]: Is it not interesting to condition the inclusion of these properties on `_PublishToBlobStorage`?

Comment (jcagme) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj]: Probably you saw a previos commit since the condition was put back?

Comment (chcosta) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj]: Oh, yup, thanks!

**Pull request merged**